### ISSUE TITLE
Sentinel Object Pattern for <<inherit>>

### ISSUE THREAD

**Issue by SchoolGuy** [cobbler/cobbler#3074]
### Is your feature request related to a problem?

Currently many of our method return or argument type definitions are `Union[str, float]` for example. This leads to confusion because in reality they don't allow setting a `str` but just a `float`. This is visible in the Docs and thus not very user friendly.

### Provide a detailed description of the proposed feature

Let's replace all usages where we allow only `<<inherit>>` and another type by the [Sentinel Object Pattern](https://python-patterns.guide/python/sentinel-object/).

### Alternatives you've considered

We could leave it like it is but this leads to confusion and duplication of code. Thus let's not let it stay like it is at the moment if time allows.

### Additional information

Originating discussion: https://github.com/cobbler/cobbler/pull/3073#discussion_r857427684

**Comment by agraul** [comment]
I guess it will match the sentinel object pattern, but it's a bit more than the naive implementation that's on [that page](https://python-patterns.guide/python/sentinel-object/) when we want useful type annotations. In this illustrative example, only the third approach covers only use case. There might be an even better way, it's just to illustrate why the naive approach is not right for us:

1. naive approach -> type annotations are completely useless
 ```python
import typing

inherit = object() # bad for type annotations, match against <class 'object'>
def foo_1() -> typing.Union[object, float]:
    return inherit
def foo_2() -> typing.Union[object, float]:
    return "foo"
```
```
# pyright
Found 1 source file
0 errors, 0 warnings, 0 infos 
```
2. slightly better approach -> type annotations are as useful as they already are
```python
import typing

inherit = "<<inherit>>" # bad for type annotations, match against <class 'str'>
def foo_3() -> typing.Union[str, float]:
    return inherit
def foo_4() -> typing.Union[str, float]:
    return "foo"
```

```
# pyright
Found 1 source file
0 errors, 0 warnings, 0 infos 
```

3. better approach -> useful type annotation
```python
import typing

T_INHERIT = typing.NewType('inherit', str)
inherit = T_INHERIT('<<inherit>>') # good for annotations, match against custom type
def foo5() -> typing.Union[T_INHERIT, float]:
    return inherit
def foo6() -> typing.Union[T_INHERIT, float]:
    return "foo"
```

```
# pyright
Found 1 source file
/home/agraul/tmp/cobbler-inherit/3.py
  /home/agraul/tmp/cobbler-inherit/3.py:9:12 - error: Expression of type "Literal['foo']" cannot be assigned to return type "inherit | float"
    Type "Literal['foo']" cannot be assigned to type "inherit | float"
      "Literal['foo']" is incompatible with "inherit"
      "Literal['foo']" is incompatible with "float" (reportGeneralTypeIssues)
1 error, 0 warnings, 0 infos 
```

I didn't put any effort into naming the type, for the PR we should come up with something a bit nicer. We'll need to think about the location where to put this as well.

**Comment by SchoolGuy** [comment]
Yes I guess the third approach is the most reasonable in my eyes. But let's see when we have time for that.
